vllm/compilation/inductor_pass.py

```diff
@@ -14,13 +19,99 @@
 logger = init_logger(__name__)
 
 
+def is_func(node: torch.fx.Node, target) -> bool:
+    return node.op == "call_function" and node.target == target
+
+
+@lru_cache(1)
+def get_hash_for_files(paths: Tuple[str, ...], extra: str = "") -> bytes:
+    """
+    Helper to compute a unique string by hashing the contents of a list of files.
+    
+    Taken from torch==2.6 (torch._inductor.custom_graph_pass.get_hash_for_files)
+    """  # noqa
+    hasher = hashlib.sha256()
+    hasher.update(extra.encode("utf-8"))
+    for path in paths:
+        with open(path, "rb") as f:
+            hasher.update(path.encode("utf-8"))
+            hasher.update(f.read())
+    return hasher.digest()
+
+
 class InductorPass(ABC):
+    """
+    General custom inductor pass interface.
+    TODO use torch._inductor.custom_graph_pass.CustomGraphPass in torch==2.6
+    """
 
     @abstractmethod
     def __call__(self, graph: torch.fx.Graph):
+        """
+        Execute the pass on the given graph.
+        """
         raise NotImplementedError
 
-    def __init__(self, config: CompilationConfig):
+    @abstractmethod
+    def uuid(self) -> Any:
+        """
+        Provide a unique identifier for the pass, used in Inductor code cache.
+        This should depend on the pass implementation, so that changes to the
+        pass result in recompilation. Use `get_hash_for_files` to hash files.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def get_hash_for_files(cls,
+                           files: Tuple[str, ...],
+                           extra: str = "") -> bytes:
+        """
+        :return: Get hash for provided files and InductorPass file.
+        """
+        return get_hash_for_files((*files, __file__), extra)
+
+
+InductorPassType: TypeAlias = Union[InductorPass, Callable[[fx.Graph], None]]
+
+
+class CallableInductorPass(InductorPass):
+
+    def __init__(self, callable: Callable[[fx.Graph], None], uuid: Any):
+        self.callable = callable
+        self._uuid = uuid
+
+    def __call__(self, graph: torch.fx.Graph):
+        self.callable(graph)
+
+    def uuid(self) -> Any:
+        return self._uuid
+
+
+def as_inductor_pass(*, uuid: Any = None, files: Tuple[str, ...] = ()):
```

Comment:
It looks like this function (and CallableInductorPass) is only used test code.  Can this code just live in the test directory?

Reply:
These functions are provided as utilities for vLLM users to add their own inductor passes and still work with the caching mechanism. I can add better comments here

Reply:
Couldn't they just inherit from `InductorPass` or whatever the appropriate class is?

Reply:
Yeah this just makes it easier, one can just slap an annotation on a function instead of rewriting it to a class. But happy to remove it if you think I should. @youkaichao would love your input on what the best way to expose custom passes to users would be.

Reply:
I think using uuid for passes is quite hacky. can we use function qualname instead?